docs/notebooks/create_synthetic_data_from_dgan_api.ipynb

```diff
@@ -0,0 +1,535 @@
+        "TIME_COLUMN = \"Date\"\n",
+        "MAX_SEQUENCE_LEN = 10\n",
+        "\n",
+        "\n",
+        "for c in COLUMNS:\n",
+        "    plt.plot(pd.to_datetime(df[TIME_COLUMN]), df[c])\n",
+        "    plt.xlabel(\"Date\")\n",
+        "    plt.ylabel(c)\n",
+        "    plt.xticks(rotation=90)\n",
+        "    plt.show()"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "Mq5UOoAokH1W",
+      "metadata": {
+        "id": "Mq5UOoAokH1W"
+      },
+      "outputs": [],
+      "source": [
+        "# Plot several 10-day sequences from real data\n",
+        "# These correspond to training examples for the DGAN model\n",
+        "def plot_subsequence(df, max_sequence_len, index):\n",
+        "    local_df = df.iloc[index * max_sequence_len:(index + 1) * max_sequence_len, :]\n",
+        "\n",
+        "    for c in COLUMNS:\n",
+        "        plt.plot(local_df[TIME_COLUMN], local_df[c], label=c)\n",
+        "    plt.legend()\n",
+        "    plt.xlabel(\"Date\")\n",
+        "    plt.ylabel(\"Price\")\n",
+        "    plt.xticks(rotation=90)\n",
+        "    plt.show()\n",
+        "\n",
+        "plot_subsequence(df, MAX_SEQUENCE_LEN, 20)\n",
+        "plot_subsequence(df, MAX_SEQUENCE_LEN, 500)\n",
+        "plot_subsequence(df, MAX_SEQUENCE_LEN, 731)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "id": "112988bf",
+      "metadata": {
+        "id": "112988bf"
+      },
+      "outputs": [],
+      "source": [
+        "# Setup config and train model\n",
+        "\n",
+        "TMP_FILE = \"tmp_train.csv\"\n",
+        "\n",
+        "project = get_project(display_name=\"DGAN-oil\", create=True)\n",
+        "\n",
+        "print(f\"Follow model training at: {project.get_console_url()}\")\n",
+        "\n",
+        "CONFIG_STRING = \"\"\"\n",
+        "schema_version: 1.0\n",
+        "name: \"dgan-oil-data\"\n",
+        "models:\n",
+        "  - timeseries_dgan:\n",
+        "        data_source: \"_\"\n",
+        "        time_column: \"Date\"\n",
+        "        df_style: \"long\"\n",
+        "        params:\n",
+        "            sample_len: 1\n",
+        "            max_sequence_len: 10\n",
+        "            feature_noise_dim: 32\n",
+        "            feature_num_layers: 1\n",
+        "            feature_num_units: 100\n",
+        "            apply_feature_scaling: True\n",
+        "            generator_learning_rate: 1e-5\n",
+        "            discriminator_learning_rate: 1e-5\n",
+        "            attribute_discriminator_learning_rate: 1e-5\n",
+        "            batch_size: 100\n",
+        "            epochs: 5000\n",
+        "            apply_example_scaling: False\n",
+        "            attribute_loss_coef: 10\n",
+        "\"\"\"\n",
+        "\n",
+        "\n",
+        "# TODO: Change to use config template once template is available.\n",
```

Comment:
Still need to update this TODO.

Reply:
We can change it with this PR, but then the notebook won't run until the config template is in the prod branch of this repo (this PR includes the config template but goes to the develop branch, so we can't test the notebook). So it's a bit awkward with the notebook and config in the same PR.

Reply:
Got it, let's do another PR after this one then?